[GStreamer] Refactor video rendering to a new element

philn opened the pull request:
#### 54cc7cad02d409128706f69a1a835f096e2beb06
<pre>
[GStreamer] Refactor video rendering to a new element
<a href="https://bugs.webkit.org/show_bug.cgi?id=240306">https://bugs.webkit.org/show_bug.cgi?id=240306</a>

Reviewed by NOBODY (OOPS!).

This new sink being based on GstVideoSink we now have processing latency, max lateness and
QoS properly configured. The sink now also advises upstream elements that they have to add a
video meta to the buffers.

* platform/GStreamer.cmake:
* platform/graphics/gstreamer/DMABufVideoSinkGStreamer.cpp:
(webKitDMABufVideoSinkConstructed):
(webKitDMABufVideoSinkGetProperty):
(webkit_dmabuf_video_sink_class_init):
(webKitDMABufVideoSinkSetMediaPlayerPrivate):
(webKitDMABufVideoSinkFinalize): Deleted.
* platform/graphics/gstreamer/GLVideoSinkGStreamer.cpp:
(webKitGLVideoSinkConstructed):
(webKitGLVideoSinkGetProperty):
(webkit_gl_video_sink_class_init):
(webKitGLVideoSinkSetMediaPlayerPrivate):
(webKitGLVideoSinkFinalize): Deleted.
* platform/graphics/gstreamer/GStreamerVideoSinkCommon.cpp: Removed.
* platform/graphics/gstreamer/GStreamerVideoSinkCommon.h: Removed.
* platform/graphics/gstreamer/LegacyVideoSinkGStreamer.cpp: Renamed from Source/WebCore/platform/graphics/gstreamer/VideoSinkGStreamer.cpp.
(VideoRenderRequestScheduler::start):
(VideoRenderRequestScheduler::stop):
(VideoRenderRequestScheduler::drain):
(VideoRenderRequestScheduler::requestRender):
(_WebKitLegacyVideoSinkPrivate::_WebKitLegacyVideoSinkPrivate):
(_WebKitLegacyVideoSinkPrivate::~_WebKitLegacyVideoSinkPrivate):
(webkitVideoSinkConstructed):
(webkitVideoSinkRepaintRequested):
(webkitVideoSinkRepaintCancelled):
(webkitVideoSinkRequestRender):
(webkitVideoSinkRender):
(webkitVideoSinkUnlock):
(webkitVideoSinkUnlockStop):
(webkitVideoSinkStop):
(webkitVideoSinkStart):
(webkitVideoSinkSetCaps):
(webkitVideoSinkProposeAllocation):
(webkitVideoSinkEvent):
(webkit_legacy_video_sink_class_init):
(webkitLegacyVideoSinkNew):
* platform/graphics/gstreamer/LegacyVideoSinkGStreamer.h: Added.
* platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp:
(WebCore::MediaPlayerPrivateGStreamer::pushTextureToCompositor):
(WebCore::MediaPlayerPrivateGStreamer::updateVideoOrientation):
(WebCore::MediaPlayerPrivateGStreamer::triggerRepaint):
(WebCore::MediaPlayerPrivateGStreamer::cancelRepaint):
(WebCore::MediaPlayerPrivateGStreamer::copyVideoTextureToPlatformTexture):
(WebCore::MediaPlayerPrivateGStreamer::createVideoSink):
* platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.h:
* platform/graphics/gstreamer/VideoSinkElementGStreamer.cpp: Added.
(sinkProposeAllocation):
(sinkShowFrame):
(sinkQuery):
(sinkEvent):
(sinkSetCaps):
(sinkGetCaps):
(sinkGetTimes):
(sinkConstructed):
(webkit_video_sink_element_class_init):
(webKitVideoSinkElementSetMediaPlayerPrivate):
(webKitVideoSinkElementSetCaps):
* platform/graphics/gstreamer/VideoSinkElementGStreamer.h: Added.
* platform/graphics/gstreamer/VideoSinkGStreamer.h: Removed.
[GStreamer] Refactor video rendering to a new element
<a href="https://bugs.webkit.org/show_bug.cgi?id=240306">https://bugs.webkit.org/show_bug.cgi?id=240306</a>

Reviewed by NOBODY (OOPS!).

* Scripts/webkitpy/style/checker.py:
</pre>

ceyusa commented on Source/WebCore/platform/graphics/gstreamer/GLVideoSinkElementGStreamer.h:
```diff
@@ -0,0 +1,60 @@
+
+G_BEGIN_DECLS
+
+#define WEBKIT_TYPE_GL_VIDEO_SINK_ELEMENT            (webkit_gl_video_sink_element_get_type ())
```
Since this element is going to be common to DMABuf and GL webkit sinks, and it doesn't do anything related with GL, perhaps it deserves another name. A plain WebKitVideoSink    perhaps?

philn replied:
That name is used for the non-GL sink already... :)

philn replied:
I could rename the slow one to "Legacy...." maybe.

zdobersek replied:
Yeah, the name's not exactly right. Otherwise it looks good.

philn committed:
[GStreamer] Refactor video rendering to a new element
https://bugs.webkit.org/show_bug.cgi?id=240306

Reviewed by NOBODY (OOPS!).

This new sink being based on GstVideoSink we now have processing latency, max lateness and
QoS properly configured. The sink now also advises upstream elements that they have to add a
video meta to the buffers.

* platform/GStreamer.cmake:
* platform/graphics/gstreamer/DMABufVideoSinkGStreamer.cpp:
(webKitDMABufVideoSinkConstructed):
(webKitDMABufVideoSinkGetProperty):
(webkit_dmabuf_video_sink_class_init):
(webKitDMABufVideoSinkSetMediaPlayerPrivate):
(webKitDMABufVideoSinkFinalize): Deleted.
* platform/graphics/gstreamer/GLVideoSinkGStreamer.cpp:
(webKitGLVideoSinkConstructed):
(webKitGLVideoSinkGetProperty):
(webkit_gl_video_sink_class_init):
(webKitGLVideoSinkSetMediaPlayerPrivate):
(webKitGLVideoSinkFinalize): Deleted.
* platform/graphics/gstreamer/GStreamerVideoSinkCommon.cpp: Removed.
* platform/graphics/gstreamer/GStreamerVideoSinkCommon.h: Removed.
* platform/graphics/gstreamer/LegacyVideoSinkGStreamer.cpp: Renamed from Source/WebCore/platform/graphics/gstreamer/VideoSinkGStreamer.cpp.
(VideoRenderRequestScheduler::start):
(VideoRenderRequestScheduler::stop):
(VideoRenderRequestScheduler::drain):
(VideoRenderRequestScheduler::requestRender):
(_WebKitLegacyVideoSinkPrivate::_WebKitLegacyVideoSinkPrivate):
(_WebKitLegacyVideoSinkPrivate::~_WebKitLegacyVideoSinkPrivate):
(webkitVideoSinkConstructed):
(webkitVideoSinkRepaintRequested):
(webkitVideoSinkRepaintCancelled):
(webkitVideoSinkRequestRender):
(webkitVideoSinkRender):
(webkitVideoSinkUnlock):
(webkitVideoSinkUnlockStop):
(webkitVideoSinkStop):
(webkitVideoSinkStart):
(webkitVideoSinkSetCaps):
(webkitVideoSinkProposeAllocation):
(webkitVideoSinkEvent):
(webkit_legacy_video_sink_class_init):
(webkitLegacyVideoSinkNew):
* platform/graphics/gstreamer/LegacyVideoSinkGStreamer.h: Added.
* platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp:
(WebCore::MediaPlayerPrivateGStreamer::pushTextureToCompositor):
(WebCore::MediaPlayerPrivateGStreamer::updateVideoOrientation):
(WebCore::MediaPlayerPrivateGStreamer::triggerRepaint):
(WebCore::MediaPlayerPrivateGStreamer::cancelRepaint):
(WebCore::MediaPlayerPrivateGStreamer::copyVideoTextureToPlatformTexture):
(WebCore::MediaPlayerPrivateGStreamer::createVideoSink):
* platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.h:
* platform/graphics/gstreamer/VideoSinkElementGStreamer.cpp: Added.
(sinkProposeAllocation):
(sinkShowFrame):
(sinkQuery):
(sinkEvent):
(sinkSetCaps):
(sinkGetCaps):
(sinkGetTimes):
(sinkConstructed):
(webkit_video_sink_element_class_init):
(webKitVideoSinkElementSetMediaPlayerPrivate):
(webKitVideoSinkElementSetCaps):
* platform/graphics/gstreamer/VideoSinkElementGStreamer.h: Added.
* platform/graphics/gstreamer/VideoSinkGStreamer.h: Removed.
[GStreamer] Refactor video rendering to a new element
https://bugs.webkit.org/show_bug.cgi?id=240306

Reviewed by NOBODY (OOPS!).

* Scripts/webkitpy/style/checker.py:
